pkg/alertmanager/alertmanager.go

```diff
@@ -241,7 +242,38 @@ func New(cfg *Config, reg *prometheus.Registry) (*Alertmanager, error) {
 		am.wg.Done()
 	}()
 
-	am.alerts, err = mem.NewAlerts(context.Background(), am.marker, 30*time.Minute, nil, am.logger)
+	var callback mem.AlertStoreCallback
+	if am.cfg.Limits != nil {
+		insertAlertFailures := promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
+			Name: "alertmanager_alerts_insert_failures_total",
+			Help: "Number of failures to insert new alerts to in-memory alert store.",
+		}, []string{"reason"})
+
+		insertAlertFailures.WithLabelValues(insertFailureTooManyAlerts)
+		insertAlertFailures.WithLabelValues(insertFailureAlertsTooBig)
+
+		limiter := newAlertsLimiter(am.cfg.UserID, am.cfg.Limits, insertAlertFailures)
```

Comment:
[nit, design] Why not passing the registered to `newAlertsLimiter` and register metrics there? I think it may lead to cleaner code.

Reply:
You're right. I guess I had some reason to do it this way, but I don't see it anymore :)